src/join.sol

```diff
@@ -318,6 +318,67 @@ contract GemJoin5 is LibNote {
     }
 }
 
+// GemJoin6
+// For a token with a proxy and implementation contract (like tUSD)
+//  If the implementation behind the proxy is changed, this prevents joins
+//   and exits until the implementation is reviewed and approved by governance.
+
+contract GemLike6 {
+    function decimals() public view returns (uint);
+    function balanceOf(address) public returns (uint256);
+    function transfer(address, uint256) public returns (bool);
+    function transferFrom(address,address,uint) public returns (bool);
+    function implementation() public view returns (address);
+}
+
+contract GemJoin6 is LibNote {
+    // --- Auth ---
+    mapping (address => uint256) public wards;
+    function rely(address usr) external note auth { wards[usr] = 1; }
+    function deny(address usr) external note auth { wards[usr] = 0; }
+    modifier auth {
+        require(wards[msg.sender] == 1, "GemJoin/not-authorized");
```

Comment:
This should be `GemJoin6/not-authorized`

Reply:
fixed